sklearn/cluster/optics_.py

```diff
@@ -544,9 +583,18 @@ def _extract_optics(ordering, reachability, maxima_ratio=.75,
     """
 
     # Extraction wrapper
-    reachability = reachability / np.max(reachability[1:])
+    # according to the paper (p. 5), for a small enough generative distance
+    # epsilong, there should be more than one INF.
+    if np.all(np.isinf(reachability)):
+        raise ValueError("All reachability values are inf. Set a larger"
+                         " max_bound.")
+    normalization_factor = np.max(reachability[reachability < np.inf])
+    reachability = reachability / normalization_factor
     reachability_plot = reachability[ordering].tolist()
-    root_node = _automatic_cluster(reachability_plot, ordering,
+    core_distances = core_distances / normalization_factor
+    core_distances_plot = core_distances[ordering].tolist()
+    root_node = _automatic_cluster(reachability_plot, core_distances_plot,
+                                   ordering,
```

Comment:
According to the paper, and our own implementation, whenever the epsilon is small enough (`max_bound`), there will be more than one `inf` in the reachability array. Also, the `_automatic_cluster` takes into account the `core_distances_` as well as the `reachability_`, hence the changes in this section.